plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/procedure/FlushHiveMetastoreCacheProcedure.java

```diff
@@ -87,20 +100,39 @@ public Procedure get()
                         new Procedure.Argument("$FAKE_FIRST_PARAMETER", VARCHAR, false, FAKE_PARAM_DEFAULT_VALUE),
                         new Procedure.Argument(PARAM_SCHEMA_NAME, VARCHAR, false, null),
                         new Procedure.Argument(PARAM_TABLE_NAME, VARCHAR, false, null),
+                        new Procedure.Argument(PARAM_PARTITION_COLUMNS, new ArrayType(VARCHAR), false, null),
+                        new Procedure.Argument(PARAM_PARTITION_VALUES, new ArrayType(VARCHAR), false, null),
                         new Procedure.Argument(PARAM_PARTITION_COLUMN, new ArrayType(VARCHAR), false, null),
                         new Procedure.Argument(PARAM_PARTITION_VALUE, new ArrayType(VARCHAR), false, null)),
                 FLUSH_HIVE_METASTORE_CACHE.bindTo(this));
     }
 
-    public void flushMetadataCache(String fakeParam, String schemaName, String tableName, List<String> partitionColumn, List<String> partitionValue)
+    public void flushMetadataCache(
+            String fakeParam,
+            String schemaName,
+            String tableName,
+            List<String> partitionColumns,
+            List<String> partitionValues,
+            List<String> partitionColumn,
+            List<String> partitionValue)
     {
         checkState(FAKE_PARAM_DEFAULT_VALUE.equals(fakeParam), "Procedure should only be invoked with named parameters. " + PROCEDURE_USAGE_EXAMPLES);
+
+        Optional<List<String>> optionalPartitionColumns = Optional.ofNullable(partitionColumns);
+        Optional<List<String>> optionalPartitionValues = Optional.ofNullable(partitionValues);
+        Optional<List<String>> optionalPartitionColumn = Optional.ofNullable(partitionColumn);
+        Optional<List<String>> optionalPartitionValue = Optional.ofNullable(partitionValue);
+        checkState(partitionParamsUsed(optionalPartitionColumns, optionalPartitionValues, optionalPartitionColumn, optionalPartitionValue)
```

Comment:
You can do all the checks within one method to avoid passing the same parameters to three different methods.

Reply:
This is the revert option to what @ksobolew asked for to increase readability :) https://github.com/trinodb/trino/pull/13566#discussion_r976517352

Reply:
Not quite.

```
        checkState(ObjectUtils.firstNonNull(partitionColumns, partitionValues) != null && ObjectUtils.firstNonNull(partitionColumn, partitionValue) != null,
                INVALID_PARTITION_PARAMS_ERROR_MESSAGE);
```

This is a bit more concise and still easy to read. Hopefully it is also correct.

Reply:
Hmm not sure. No strong opinion one or another side.
On one hand i like that method names are self explanatory now on the other hand your solution will have a little less code.  Let @ksobolew give his opinion -> WDYT ?

Reply:
Well, `firstNonNull(partitionColumns, partitionValues) != null` is not really transparent what it means, IMO

Reply:
This is not the most important thing in the world though

Reply:
> Well, firstNonNull(partitionColumns, partitionValues) != null is not really transparent what it means,

This can be wrapped in a helper method and then it makes more sense

```
private static boolean isAnyPresent(Object o1, Object o2) 
{
     return firstNonNull(o1, o2) != null
}
```

Reply:
But then we end up in same state 
```
checkState(!isAnyPresent(partitionColumns, partitionValues) && !isAnyPresent(partitionColumn, partitionValue) 
   || isAnyPresent(partitionColumns, partitionValues) && !isAnyPresent(partitionColumn, partitionValue) 
   || !isAnyPresent(partitionColumns, partitionValues) && isAnyPresent(partitionColumn, partitionValue),
   "Error msg")
```
Which still would required wrapping them with some method giving context to avoid confusion.
As I still use Optionals later on and they are useful I think I will leave it as it is.